Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -0,0 +1,215 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// Controller responsible for displaying the channel messages.
+@available(iOSApplicationExtension, unavailable)
+open class ChatChannelVC: _ViewController, ThemeProvider {
+    /// User search controller passed directly to the composer
+    open var userSuggestionSearchController: ChatUserSearchController!
+
+    /// Controller for observing data changes within the channel
+    open var channelController: ChatChannelController!
+
+    public var client: ChatClient {
+        channelController.client
+    }
+
+    /// Component responsible for setting the correct offset when keyboard frame is changed.
+    open lazy var keyboardHandler: KeyboardHandler = ComposerKeyboardHandler(
+        composerParentVC: self,
+        composerBottomConstraint: messageComposerBottomConstraint
+    )
+
+    open lazy var messageListVC: ChatMessageListVC = components
+        .messageListVC
+        .init()
+
+    /// Controller that handles the composer view
+    open private(set) lazy var messageComposerVC = components
+        .messageComposerVC
+        .init()
+
+    private var messageComposerBottomConstraint: NSLayoutConstraint?
+
+    /// Header View
+    open private(set) lazy var headerView: ChatChannelHeaderView = components
+        .channelHeaderView.init()
+        .withoutAutoresizingMaskConstraints
+
+    /// View for displaying the channel image in the navigation bar.
+    open private(set) lazy var channelAvatarView = components
+        .channelAvatarView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+
+        messageListVC.delegate = self
+        messageListVC.dataSource = self
+        messageListVC.client = client
+
+        messageComposerVC.channelController = channelController
+        messageComposerVC.userSearchController = userSuggestionSearchController
+
+        channelController.setDelegate(self)
+        channelController.synchronize { [weak self] _ in
+            self?.messageComposerVC.updateContent()
+        }
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        view.backgroundColor = appearance.colorPalette.background
+
+        messageListVC.view.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
this is repeating some N times across the code, `translatesAutoresizingMaskIntoConstraints = false` 
lets put this code into the private UIView extension and make it more sugar
```Swift
private extension UIView {
   func useAutoLayout() {
       translatesAutoresizingMaskIntoConstraints = false
   }
}
```
then it will be `view.useAutoLayout()` instead of `view.translatesAutoresizingMaskIntoConstraints = false`

Reply:
We already have a helper for this. `withoutAutoresizingMaskConstraints`. In most of the codebase, we are using the helper, in some parts we are not, it depends on where it makes sense.

Reply:
Okay than this could be used, also ideas how to make a DRY-ied code
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

```Swift
[firstView, secondView, thirdView].forEach {
   $0. withoutAutoresizingMaskConstraints
}
```

This will decrease amount of LOCs -> decrease cost of support / amount of possible bugs

Reply:
also upper solution with array also can have a nice sugared extension with specified `where` type for Array generic 
and in result will be
```Swift
[firstView, secondView, thirdView].withoutAutoresizingMaskConstraints()
```

Reply:
We do have DRY coding (hence the need for this PR), but right now we want to focus mainly on architectural problems. Don't think we should focus on small details like this. Also +LOCs != Cost of Support. Code should be clear and readable, and normally that means +LOCs.

Reply:
Making copy-paste code = have problems in near future, please DRY PR as much as possible. 
We should focus on every detail we have cause we making open-source product which should show good approach to customers/other our code readers.

Reply:
You are asking to change `messageComposerVC.view.translatesAutoresizingMaskIntoConstraints = false` -> `messageComposerVC.view.withoutAutoresizingMaskConstraints()` . It is replacing one line, with another one line. Also currently `withoutAutoresizingMaskCOnstraints` returns the view itself, so I will need to do `_ = messageComposerVC.view.withoutAutoresizingMaskConstraints` which will look even worse. (And no I can't use @discardableResult because it is a property). So in order to fix this, I will need to convert `withoutAutoresizingMaskConstraints` into a function instead which will cause a lot of changes in an already huge PR. 

Btw, DRY is not always a good thing: [Why DRY principle is bad advice](https://rotemtam.com/2020/05/18/the-dry-principle-is-bad-advice/).

Reply:
I am asking to do generic extension for array where object is UIView and make a 
>[firstView, secondView, thirdView].withoutAutoresizingMaskConstraints()

in current case it will be

```
messageListVC.view.translatesAutoresizingMaskIntoConstraints = false
messageComposerVC.view.translatesAutoresizingMaskIntoConstraints = false
```
VS

`[messageListVC.view, messageComposerVC.view].withoutAutoresizingMaskConstraints()`

and continue to use this array extension in future
for now it 2 line vs 1 line, in future it will N lines vs 1 line

Reply:
`withoutAutoresizingMaskConstraints` is a property, not a function. As I said, changing to a function will do a lot of changes to the codebase and increasing an already 1000+ lines PR to even more. Feel free to create a ticket and address this issue 🙂

Reply:
Okay, lets put this to backlog then, 
I will show after, this will be a change for ~10 LOCs

Reply:
https://stream-io.atlassian.net/browse/CIS-1133